doc/languages-frameworks/qt.section.md

```diff
@@ -11,23 +11,20 @@ pure and explicit at build-time, at the cost of introducing an extra indirection
 ## Nix expression for a Qt package (default.nix) {#qt-default-nix}
 
 ```nix
-{ stdenv, lib, qtbase, wrapQtAppsHook }:
+{ stdenv, lib, qt6, wrapQtAppsHook }:
 
 stdenv.mkDerivation {
   pname = "myapp";
   version = "1.0";
 
-  buildInputs = [ qtbase ];
-  nativeBuildInputs = [ wrapQtAppsHook ];
+  buildInputs = [ qt6.qtbase ];
+  nativeBuildInputs = [ qt6.wrapQtAppsHook ];
 }
 ```
 
-It is important to import Qt modules directly, that is: `qtbase`, `qtdeclarative`, etc. *Do not* import Qt package sets such as `qt5` because the Qt versions of dependencies may not be coherent, causing build and runtime failures.
-
-Additionally all Qt packages must include `wrapQtAppsHook` in `nativeBuildInputs`, or you must explicitly set `dontWrapQtApps`.
+Any Qt package should include `wrapQtAppsHook` in `nativeBuildInputs`, or explicitly set `dontWrapQtApps` to bypass generating the wrappers.
 
-`pkgs.callPackage` does not provide injections for `qtbase` or the like.
-Instead you want to either use `pkgs.libsForQt5.callPackage`, or `pkgs.qt6Packages.callPackage`, depending on the Qt version you want to use.
+If your package can be built with multiple Qt versions, you may want to take Qt modules as separate arguments (`qtbase`, `qtdeclarative` etc.), and then instantiate the package with `pkgs.libsForQt5.callPackage`, or `pkgs.qt6Packages.callPackage`, depending on the Qt version you want to use.
```

Comment:
But where should they place the new packages in this kind?

Reply:
In the old structure, at least for now.

Reply:
Theoretically possible, but the sets would never be consistent, because there are major API changes between 5 and 6.

Reply:
> In the old structure, at least for now.

As far as I'm concerned, we shouldn't depend on the old structure from now on. If it belongs to a package set or scope or has special dependencies, there may be exceptions, but there are too many applications that support both qt5 and qt6. For packages already in Nixpkgs, we should also make treewide changes after the automatic migration is completed so that they can be migrated to the by-name directory structure.

Reply:
Are there really that many? And do we really need to shove them all into by-name?

Reply:
> Are there really that many? 

It seems that not many of them already support qt5 and qt6 in nixpkgs, but there may be many that "should support" but don't.

> And do we really need to shove them all into by-name?

Maybe ask @infinisil. The RFC and the guide don't say that it must be placed in by-name, but perhaps we should still try to minimize this.

Reply:
I generally don't think we should be having two versions of every single package that _can_ be built with both Qt versions. It's fine to use whatever is preferred upstream, and multiple versions can be added if there's actual demand for them and the maintenance costs aren't too much.

Reply:
For top-level attrs, I'd rather discourage these custom callPackages entirely, as all they're really doing is masking which package set the expression args come from.

Reply:
That's what the `by-name` structure already achieves.

Reply:
using `by-name` doesn't prevent a custom `callPackage` though